src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java

```diff
@@ -155,33 +155,43 @@ public synchronized void addRow(Object... values) throws SQLServerException {
      */
     private void internalAddrow(JDBCType jdbcType, Object val, Object[] rowValues,
             Map.Entry<Integer, SQLServerDataColumn> pair) throws SQLServerException {
+        if (null != val) {
+            SQLServerDataColumn currentColumnMetadata = pair.getValue();
+            int nValueLen;
+            switch (jdbcType) {
+                case BIGINT:
+                    rowValues[pair.getKey()] = (val instanceof Long) ? (long) val : Long.parseLong(val.toString());
```

Comment:
Rather than casting from the Object to the primitive, it would be better to keep this as the object itself since it is going to be assigned directly into an Object[].  That will prevent unnecessary unboxing and reboxing.

That is to say, this line could read

    (val instanceof Long) ? val : Long.parseLong(val.toString())